activestorage/app/models/active_storage/variation.rb

```diff
@@ -68,9 +68,9 @@ def image_processing_transform(file, format)
       operations = transformations.inject([]) do |list, (name, argument)|
         if name.to_s == "combine_options"
           ActiveSupport::Deprecation.warn("The ImageProcessing ActiveStorage variant backend doesn't need :combine_options, as it already generates a single MiniMagick command. In Rails 6.1 :combine_options will not be supported anymore.")
-          list.concat argument.to_a
+          list.concat argument.keep_if { |key, value| value.present? }.to_a
         else
-          list << [name, argument]
+          list.tap { |list| list << [name, argument] if argument.present? }
```

Comment:
There’s no need for `tap` here. Move the `argument` condition to the `else` statement above:

```ruby
if name.to_s == "combine_options"
  # ...
elsif argument.present?
  list << [name, argument]
end
```

Reply:
I ended up moving the line to `list << [name, argument]` like it was previously, but I had to wrap the entirety of the `inject` block in a `list.tap` call to guard against the new case where neither conditional fires. Without that, it implicitly returns `nil`, which causes the `inject` method to fail.